### PR TITLE
Modify User model to store email

### PR DESCRIPTION
**Pull request by gaoxk**
## Notion ticket link

[Modify User model to store email](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=ba22ac93d09040ff8219c60ac7caabdb&p=9d2011f9d77945ecb7bd6f6f7926056f)

## Implementation description

- add email column to user table

## Steps to test

1. start with an empty db (easiest way to do this is `docker-compose down --volumes` to discard the db volumes) 
2. `docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db upgrade"`
3. verify that the email col exists 
```
docker exec -it planet-read_db_1 /bin/bash
mysql -u root -proot
USE planet-read;
SELECT * FROM users;
```
4. `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data"`
5. log in and poke around. verify things are working

## What should reviewers focus on?

- does it work!!?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by jennchenn — CHANGES_REQUESTED** [review]
I ran into some issues running the upgrade command on my database (I pasted the error in a comment below)

Also for the testing, I think running on the existing up-to-date database mimics our use case more, since I ideally (I think) we would want the versions to allow us to roll the database up without needing to start from a completely empty database (the heroku deployments are set up right now to just upgrade on merge without needing to downgrade to base first)? I get the same error on an empty database though

File: backend/python/migrations/versions/b803e4c97ed1_add_email_to_user_table.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('users', sa.Column('email', mysql.TEXT(), nullable=True))
+    op.create_unique_constraint(None, 'users', ['email'])
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_constraint(None, 'users', type_='unique')
```
[the remark this answers is not in the document]
Comment (gaoxk) [reply backend/python/migrations/versions/b803e4c97ed1_add_email_to_user_table.py:28]: bah, I didnt attempt a downgrade in my testing. downgrading wasnt working, I've added the constraint name now.

**Review by jennchenn — CHANGES_REQUESTED** [review]
ahh not sure why the migrations don't seem to work properly on my machine with the way you have it currently implemented

on another note, we will need to empty the database of test data before deploying this branch or else it'll fail to upgrade since the email column is non-nullable and also has a unique constraint (if you think we will be making more changes similar to this, would it be beneficial to re-order the commands in the `Procfile` to clear data->upgrade->insert data->start server? i.e.:
```
web: flask db upgrade & wait -n; python -m tools.insert_test_data erase & wait -n; python -m tools.insert_test_data & wait -n; python server.py
``` 
to
```
web: python -m tools.insert_test_data erase & wait -n; flask db upgrade & wait -n; python -m tools.insert_test_data & wait -n; python server.py
```

**Review by jennchenn — APPROVED** [review]
looks good to me!

**Pull request merged**